test/integration/controller-manager/matching_test.go

```diff
@@ -171,16 +171,16 @@ func TestMatching(t *testing.T) {
 	}
 	expectation := map[gvrnn]any{}
 	for _, obj := range append(nsORs, objsCreated...) {
-		key, err := util.KeyForGroupVersionKindNamespaceName(obj.MRObject)
+		identifier, err := util.RuntimeObjectToObjectIdentifier(obj.MRObject, ctlr.GetGvkToGvrMapper())
 		if err != nil {
 			t.Fatalf("Failed to extract Key from %#v: %s", obj.MRObject, err)
 		}
-		k2 := k2r(key, obj.Resource)
+		id2 := id2r(identifier, obj.Resource)
```

Comment:
FYI: `identifier` already carries the resource, no need for this func to take a second argument.